pilot/cmd/pilot-agent/status/server.go

```diff
@@ -165,6 +177,7 @@ func (s *Server) Run(ctx context.Context) {
 	mux.HandleFunc(`/stats/prometheus`, s.handleStats)
 	mux.HandleFunc(quitPath, s.handleQuit)
 	mux.HandleFunc("/app-health/", s.handleAppProbe)
+	mux.Handle("/agent-health", promStatsExporter)
```

Comment:
Instead of doing this, can you just update `s.handleStats` just call `promStatsExporter.ServeHTTP` ?

Reply:
this was my initial thought. However, I don't know how I can get the response from the `promStatsExporter` because `ServeHTTP` does not give me anything.

The alternative was to use an `httptest` responsewriter recorder to capture the response, but im not sure its appropriate to use a test lib in here.

Reply:
I think if you pass w here then the prom http handled will write to w which is what we need?

Reply:
oh shoot i misunderstoof what you were saying yeah ok got it